Bottom: 5e4767b668fa63e23e92250a8f0a87d024ac6aec
Top:    906019518445e465769d6b65a2c073cdfc3e242f
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-12 08:59:16 -0500

Refresh of tmp-update-gitlab-version

---
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index 9e2f0b60..82072b97 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -57,13 +57,6 @@
     owner: root
     group: root
     mode: 'u=rw,g=r,o='
-
-# - name: copy env-file
-#   command: 'cp --verbose --update --backup=numbered {{ docker_projects_dir }}/docker-gitlab/gitlab.env.list {{ gitlab_config_dir }}'
-
-# - name: copy script
-#   command: 'cp --verbose --update --backup=numbered {{ docker_projects_dir }}/docker-gitlab/gitlab.sh {{ gitlab_config_dir }}'
-
 - name: exists - state file
   stat:
     path: '{{ gitlab_config_dir }}/restore.date.txt'
@@ -71,30 +64,6 @@
     get_md5: False
   register: st_gitlab_restore
 
-# *****************************************************************************
-# Update or make the image.
-
-# - name: Checkout image repo
-#   git:
-#     repo: '{{ gitlab_image_repo }}'
-#     version: master
-#     dest: '{{ docker_projects_dir }}/docker-gitlab'
-
-# - name: build image
-#   docker_image:
-#     name: '{{ gitlab_image_name }}'
-#     tag: '{{ docker_image_tag }}'
-#     path: '{{ docker_projects_dir }}/docker-gitlab'
-#     force: "{{ docker_image_force_build }}"
-
-# Data volume image is now the same as the container image
-#- name: build image
-#  docker_image:
-#    image_name: '{{ gitlab_dv_image_name }}'
-#    image_tag: '{{ docker_image_tag }}'
-#    dockerfile_dir: '{{ docker_projects_dir }}/docker-gitlab'
-#    force: "{{ docker_image_force_build }}"
-
 # *****************************************************************************
 # Create the data volumes
 
@@ -107,13 +76,16 @@
   docker_volume:
     name: '{{ gitlab_dv_name }}'
 
+# *****************************************************************************
+# Create the gitlab container
+
 - name: create container (gitlab)
   docker_container:
     name: '{{ gitlab_container_name }}'
     detach: true
     restart_policy: '{{ docker_restart_policy }}'
-    hostname: '{{ gitlab_network.hostname }}'
-    ports: '{{ gitlab_network.port_args }}'
+    hostname: '{{ gitlab_hostname }}'
+    ports: '{{ gitlab_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     volumes:
diff --git a/roles/docker-gitlab/tasks/restore.yml b/roles/docker-gitlab/tasks/restore.yml
index 38cbfae8..aeeef7a4 100644
--- a/roles/docker-gitlab/tasks/restore.yml
+++ b/roles/docker-gitlab/tasks/restore.yml
@@ -5,8 +5,9 @@
 # Get data from tape
 
 - name: find files
-  files_in_dir:
+  find:
     path: '{{ gitlab_docker_restore_dir }}'
+    file_type: any
   register: gitlab_backup
 
 - name: data from tape
@@ -16,11 +17,13 @@
     fileset: '{{ bacula_fileset }}'
     dest: '{{ bacula_dest }}'
     path_to_restore: '{{ gitlab_docker_backup_dir }}'
-  when: gitlab_backup.file_list == []
+  when: gitlab_backup.matched == 0
 
 - name: find files
-  files_in_dir:
+  find:
     path: '{{ gitlab_docker_restore_dir }}'
+    file_type: any
+    follow: yes
   register: gitlab_backup
 
 - name: File permissions
@@ -35,20 +38,49 @@
 # *****************************************************************************
 # restore the gitlab backup
 
-- name: Copy backup file to container
+- name: Copy backup files to container
   copy:
     remote_src: yes
-    src: '{{ gitlab_docker_restore_dir }}/{{ gitlab_backup.file_list | first}}'
+    src: '{{ item.path }}'
     dest: '{{ gitlab_docker_backup_dir }}/'
     owner: root
     group: root
     mode: 'u=rwx,g=rwx,o=rwx'
+  loop: "{{ gitlab_backup.files }}"
+
+- name: shut down gitlab services for restoration
+  command: >
+    docker exec -t {{ gitlab_container_name }} gitlab-ctl stop {{ item }}
+  loop:
+  - unicorn
+  - sidekiq
+
+- name: gitlab restore script
+  command: >
+    docker exec -t {{ gitlab_container_name }}
+    /bin/sh -c 'tar vxzf /mnt/backups/etc-gitlab-*.tgz -C /' 
 
 - name: gitlab restore script
   command: >
-    docker exec -t {{ gitlab_container_name }} /bin/sh -c
-    'gitlab-ctl stop unicorn && gitlab-ctl stop sidekiq && gitlab-rake gitlab:backup:restore force=yes'
-  when: gitlab_backup.file_list != []
+    docker exec -t {{ gitlab_container_name }}
+    gitlab-rake gitlab:backup:restore force=yes
+
+- name: reconfigure gitlab for updated configuration
+  command: >
+    docker exec -t {{ gitlab_container_name }} gitlab-ctl reconfigure
+
+- name: restart gitlab services
+  command: >
+    docker exec -t {{ gitlab_container_name }} gitlab-ctl start
+
+- name: Set root user password in restored database
+  command: >
+    docker exec -t {{ gitlab_container_name }} 
+    gitlab-rails runner 
+      'user=User.where(id: 1).first;
+       user.password="{{ gitlab_root_password }}";
+       user.password_confirmation="{{ gitlab_root_password }}";
+       user.save!'
 
 # *****************************************************************************
 # cleanup
diff --git a/roles/docker-gitlab/templates/gitlab.rb.j2 b/roles/docker-gitlab/templates/gitlab.rb.j2
index 39984189..8102b4bb 100644
--- a/roles/docker-gitlab/templates/gitlab.rb.j2
+++ b/roles/docker-gitlab/templates/gitlab.rb.j2
@@ -10,7 +10,7 @@
 ##! URL on which GitLab will be reachable.
 ##! For more details on configuring external_url see:
 ##! https://docs.gitlab.com/omnibus/settings/configuration.html#configuring-the-external-url-for-gitlab
-external_url 'https://{{ git_hostname }}'
+external_url 'https://{{ gitlab_hostname }}'
 
 ## Roles for multi-instance GitLab
 ##! The default is to have no roles enabled, which results in GitLab running as an all-in-one instance.
@@ -45,7 +45,7 @@ external_url 'https://{{ git_hostname }}'
 ## gitlab.yml configuration
 ##! Docs: https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/doc/settings/gitlab.yml.md
 ################################################################################
-gitlab_rails['gitlab_ssh_host'] = '{{ git_hostname }}'
+gitlab_rails['gitlab_ssh_host'] = '{{ gitlab_hostname }}'
 gitlab_rails['time_zone'] = 'America/Chicago'
 
 ### Email Settings
@@ -200,7 +200,7 @@ gitlab_rails['ldap_enabled'] = true
 gitlab_rails['ldap_servers'] = YAML.load <<-'EOS'
   main: # 'main' is the GitLab 'provider ID' of this LDAP server
     label: 'NovaTech'
-    host: '{{ ldap_hostname }}'
+    host: '{{ openldap_hostname }}'
     port: 389
     uid: 'uid'
     bind_dn: 'cn=proxyagent,dc=novatech'
